### PR TITLE
Set active tutorial depending on UPP state

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-classifier

Describe your changes:
Sets up an observer in the TutorialStore and updates seen date time depending on the UPP state. Still WIP due to tests. I removed the seen time for mini-courses because I refreshed myself on how these work and they are only shown to logged in users, so we do not need to track the seen time for the session in addition to the UPP. 

The store is getting a bit unwieldy, so I think the common actions between tutorials and minicourses should be pulled out into a general store, then a store for tutorial kind and minicourse kind could be composed as a follow up, post TESS launch. 

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by srallen** [comment]
@eatyourgreens do you have any suggestions on getting some async action tests working? Starting on line 468, I'm trying to test the action to show the tutorial as a modal and the expectations are failing because they are getting called before the actions are finishing and I'm not sure what I'm doing wrong. I've added sinon spies as well as console logs to see that the tests are failing but the actions I'm expecting to be called are being called since the logs show up. These few broken tests are the only thing holding up this PR.

**Comment by eatyourgreens** [comment]
I’m not sure but maybe begin the test with `fetchTutorials()` instead of `projects.setActive()`? I’ll have a closer look tomorrow but I think the test is running when the project is ready rather than waiting for the UPP and tutorial.

**Comment by eatyourgreens** [comment]
Another thought: if the problem is a chain of async actions, and the expectations are running after the first and not waiting for the second, then maybe break the tests up into tests for each async action resolving in turn:

On async action 1, expect async action 2 to start. 

On async action 2, expect the spies to be called.

**Comment by srallen** [comment]
It dawned to me what was going wrong, so there's now working tests. One of the psot-TESS things I'd like to do is clean up how we're mocking the client and incorporate that into panoptes.js itself so we're consistent between how we're doing that. I'm not really happy with how it's been done so far since it's a bit hard to follow and make sure you're doing it right, but it's working.

**Comment by eatyourgreens** [comment]
This is crashing the project app when I try to run it at http://localhost.zooniverse.org:3000/projects/eatyourgreens/-project-testing-ground/classify?env=staging `Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: null.`

**Comment by eatyourgreens** [comment]
As an anonymous user, should I see the tutorial open in a modal when I load the classify page? I'm not seeing that happen.

**Comment by srallen** [comment]
@eatyourgreens I should've been more clear. This PR is only the store updates for setting the active state and timestamp. I will have a follow up PR to add the actual UI component that shows the tutorial in a modal.

**Comment by eatyourgreens** [comment]
I've been stubbing client methods for tests but that can be tedious too.
https://github.com/zooniverse/front-end-monorepo/blob/ff89e8f68858d4ea8cf64bf2b974ada5ba21cdd6/packages/app-project/stores/Collections.spec.js#L31-L33

Incidentally, the project app's initStore helper is useful for setting up tests.
https://github.com/zooniverse/front-end-monorepo/blob/ff89e8f68858d4ea8cf64bf2b974ada5ba21cdd6/packages/app-project/stores/Collections.spec.js#L29-L30

**Comment by eatyourgreens** [comment]
> The store is getting a bit unwieldy, so I think the common actions between tutorials and minicourses should be pulled out into a general store, then a store for tutorial kind and minicourse kind could be composed as a follow up, post TESS launch.

In the Collections store, I moved common code for favourites & collections into a couple of private functions (one to fetch collections and another to create a new collection.) The public actions then became simple wrappers for those functions, and it's only the public actions that are tested.

That pattern might work for tutorials/minicourses too, if you're making the same API calls but with different queries.
